### PR TITLE
Regular update (2023/03)

### PR DESCRIPTION
**Pull request by masa213f**
Signed-off-by: Masayuki Ishii <masa213f@gmail.com>

### CODE REVIEW

File: grafana/Dockerfile [hunk 1]
```diff
@@ -55,14 +55,13 @@ RUN mkdir -p "$GF_PATHS_HOME/.aws" && \
              "$GF_PATHS_LOGS" \
              "$GF_PATHS_PLUGINS" \
              "$GF_PATHS_DATA" && \
-    cp "$GF_PATHS_HOME/conf/sample.ini" "$GF_PATHS_CONFIG" && \
-    cp "$GF_PATHS_HOME/conf/ldap.toml" /etc/grafana/ldap.toml && \
+    cp "conf/sample.ini" "$GF_PATHS_CONFIG" && \
+    cp "conf/ldap.toml" /etc/grafana/ldap.toml && \
     chown -R 10000:10000 "$GF_PATHS_DATA" "$GF_PATHS_HOME/.aws" "$GF_PATHS_LOGS" "$GF_PATHS_PLUGINS" "$GF_PATHS_PROVISIONING" && \
-    chmod 777 "$GF_PATHS_DATA" "$GF_PATHS_HOME/.aws" "$GF_PATHS_LOGS" "$GF_PATHS_PLUGINS" "$GF_PATHS_PROVISIONING"
+    chmod -R 777 "$GF_PATHS_DATA" "$GF_PATHS_HOME/.aws" "$GF_PATHS_LOGS" "$GF_PATHS_PLUGINS" "$GF_PATHS_PROVISIONING"
```
Comment (masa213f) [inline grafana/Dockerfile:61]: same as https://github.com/grafana/grafana/blob/v9.4.3/Dockerfile#L147-L150

File: grafana/Dockerfile [hunk 2]
```diff
@@ -55,14 +55,13 @@ RUN mkdir -p "$GF_PATHS_HOME/.aws" && \
...
 
 COPY --from=build /work/grafana/bin/*/grafana-server /work/grafana/bin/*/grafana-cli ./bin/
 COPY --from=build /work/grafana/public ./public
-COPY --from=build /work/grafana/tools ./tools
```
Comment (masa213f) [inline grafana/Dockerfile:65]: tools directory is empty.
```
$ docker run -it --rm --entrypoint=bash quay.io/cybozu/grafana:9.1.4.1
grafana@0c67313df0ce:/usr/share/grafana$ ls -l tools
total 0
```

File: memcached/Dockerfile [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
+ARG MEMCACHED_VERSION=1.6.19
 
 WORKDIR /work
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 RUN apt-get update \
     && apt-get install -y --no-install-recommends \
         automake \
-        libevent-dev
+        libevent-dev \
+        pkg-config
```
Comment (masa213f) [inline memcached/Dockerfile:14]: autoconf depends on pkg-config.
https://github.com/memcached/memcached/issues/932

**Comment by umezawatakeshi** [comment]
How about promtail?

**Comment by masa213f** [comment]
@umezawatakeshi 
I skipped it because `no-newline-while-joining-partial-lines.patch` was not patched to promtail v2.7.4.
I will update it later.

The patch conflicts with the following change.
https://github.com/grafana/loki/commit/234dcca9ab11ed08cc91b28466c7700806753d2c


```
#0 0.327 2 out of 2 hunks FAILED -- saving rejects to file clients/pkg/logentry/stages/extensions.go.rej
------
Dockerfile:65
--------------------
  63 |     # see https://github.com/grafana/loki/issues/7996
  64 |     COPY no-newline-while-joining-partial-lines.patch /work/
  65 | >>> RUN cd /work/loki && patch -p1 < ../no-newline-while-joining-partial-lines.patch
  66 |     RUN make -C /work/loki BUILD_IN_CONTAINER=false promtail
  67 |     
--------------------
ERROR: failed to solve: process "/bin/bash -o pipefail -c cd /work/loki && patch -p1 < ../no-newline-while-joining-partial-lines.patch" did not complete successfully: exit code: 1
```

**Comment by masa213f** [comment]
I checked the history. The patch was no longer needed.
So I will remove the patch and update promtail to v2.7.4. 62c109a

**Review by umezawatakeshi — APPROVED** [review]

**Pull request merged**